lib/brew_dash/schema/brew.ex

```diff
@@ -49,8 +55,14 @@ defmodule BrewDash.Schema.Brew do
       :notes,
       :batch_number,
       :original_gravity,
-      :final_gravity
+      :final_gravity,
+      :source,
+      :source_id,
+      :source_recipe_id
     ])
     |> validate_required([:status])
   end
+
+  def changeset_with_source(brew, attrs),
+    do: brew |> changeset(attrs) |> unique_constraint([:source, :source_id])
```

Comment:
See my comment in `Recipe` about the source being case only in the `changeset_with_source`.